File: 1. 단계별로_풀어보기/10~19/16. 스택, 큐, 덱 1/6. 18258.py
from collections import deque
import sys

input = sys.stdin.readline
output = sys.stdout.write

N = int(input())

queue = deque()

for _ in range(N):
  query = list(input().strip().split(' '))

  order = query[0]

  if order == 'push':
    X = int(query[1])
    queue.append(X)
  elif order == 'pop':
    if len(queue) > 0:
      output(f"{queue.popleft()}\n")
    else:
      output('-1\n')
  elif order == 'size':
    output(f"{len(queue)}\n")
  elif order == 'empty':
    output(f"{1 if len(queue) == 0 else 0}\n")
  elif order == 'front':
    if len(queue) > 0:
      output(f"{queue[0]}\n")
    else:
      output("-1\n")
  else:
    if len(queue) > 0:
      output(f"{queue[-1]}\n")
    else:
      output("-1\n")
  

File: 1. 단계별로_풀어보기/10~19/16. 스택, 큐, 덱 1/6. 18258.js
// 입력
const fs = require('fs')
const inputArr = fs.readFileSync(0).toString().trim().split('\n')

// 입력 파싱
const N = +inputArr[0]
const queryArr = inputArr.slice(1).map(row => row.trim().split(' '))

// 큐 역할을 할 배열 선언
const queue = Array.from({length: 2000000}).fill(0)
let headIdx = 0
let tailIdx = -1

// 출력할 결과 변수
let result = ''

queryArr.forEach(query => {
  const order = query[0]
  const isEmpty = headIdx > tailIdx

  if (order === 'push') {
    const X = +query[1]

    queue[++tailIdx] = X
  } // ::
  else if (order === 'pop') {
    if (isEmpty) {
      result += '-1\n'
    } // ::
    else {
      result += queue[headIdx++] + '\n'
    }
  } // ::
  else if (order === 'size') {
    const sizes = tailIdx - headIdx + 1
    result += sizes + '\n'
  } // ::
  else if (order === 'empty') {
    result += (isEmpty ? 1 : 0) + '\n'
  } // ::
  else if (order === 'front') {
    if (isEmpty) {
      result += '-1\n'
    } // ::
    else {
      result += queue[headIdx] + '\n'
    }
  } // ::
  else {
    if (isEmpty) {
      result += '-1\n'
    } // ::
    else {
      result += queue[tailIdx] + '\n'
    }
  }
})

// 출력
process.stdout.write(result)
